### PR TITLE
Visualise excelwork book sheets, db tables/schemas and add node adding gets to query them 

### PR DESCRIPTION
**Pull request by marthasharkey**
### Pull Request Description
changes to current table (and db table) on index click:
- node added on double click 
- tooltip added
- cursor changes to pointer 
- styled as a link

7473-table-w-links

Row table types:
Added on click to show value 
7473-row-links

Excel Workbook connections
- show sheet names in column 
- sheet names are clickable that add a 'read' node for the corresponding sheet 
- Shown in table with header title: Sheets 

7473-excel-links-sheets

SQLite Connection:
- shows available schemas in column 
- schemas are clickable that add a 'query' node for the corresponding schema/table 
7473-sql-lite-links

Postgres Connection:
- shows available tables in a column 
- tables are clickable and add a 'query' node for the corresponding table 
7473-postgres-links

JS_Object
- style keys as links 
7473-json-links

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/gui2/src/components/visualizations/TableVisualization.vue [hunk 1]
```diff
@@ -296,23 +379,32 @@ const getTablePattern = (index: number) =>
       ),
     ),
   )
+
 function createNode(params: CellClickedEvent) {
-  if (config.nodeType === VECTOR_NODE_TYPE || config.nodeType === COLUMN_NODE_TYPE) {
+  console.log(config.nodeType)
```
Comment (GregoryTravis) [inline app/gui2/src/components/visualizations/TableVisualization.vue]: Is this just for debugging?

**Review by GregoryTravis — APPROVED** [review]

File: app/gui2/src/components/visualizations/TableVisualization.vue (hunk 1 above)
Comment (marthasharkey) [reply app/gui2/src/components/visualizations/TableVisualization.vue]: yes thank you for spotting this!

File: app/gui2/src/components/visualizations/TableVisualization.vue [hunk 2]
```diff
@@ -158,6 +174,70 @@ const selectableRowLimits = computed(() => {
...
+    case SQLITE_CONNECTIONS_NODE_TYPE:
+    case POSTGRES_CONNECTIONS_NODE_TYPE:
+      val = 'table'
+      break
+    case TABLE_NODE_TYPE:
+    case DB_TABLE_NODE_TYPE:
+      val = 'row'
+  }
+  return val
+})
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/TableVisualization.vue]: The logic might be easier to follow with one computed/switch selecting between different constant objects, with each object having fields like `selector`/`identifierAction`/`tooltipValue`. Up to you.

**Review by kazcw — APPROVED** [review]

File: app/gui2/src/components/visualizations/TableVisualization.vue (hunk 2 above)
Comment (radeusgd) [reply app/gui2/src/components/visualizations/TableVisualization.vue]: +1 I wanted to write the same thing

File: app/gui2/src/components/visualizations/TableVisualization.vue [hunk 3]
```diff
@@ -296,23 +379,31 @@ const getTablePattern = (index: number) =>
...
 }
 
-function indexField(): ColDef {
-  return { field: INDEX_FIELD_NAME, onCellClicked: (params) => createNode(params) }
+function toLinkField(fieldName: string): ColDef {
+  return {
+    field: fieldName,
+    onCellDoubleClicked: (params) => createNode(params),
+    tooltipValueGetter: () => {
+      return `Double click to view this ${onClickTooltipValue.value} in seperate node`
```
Comment (radeusgd) [inline app/gui2/src/components/visualizations/TableVisualization.vue]:
```suggestion
      return `Double click to view this ${onClickTooltipValue.value} in a seperate node`
```

nitpick: I'm not 100% sure but shouldn't there be an article here?

File: test/Visualization_Tests/src/Table_Spec.enso [hunk 4]
```diff
@@ -36,6 +36,12 @@ type Foo
     to_js_object : JS_Object
     to_js_object self = JS_Object.from_pairs [["x", self.x]]
 
+type FooLink
```
Comment (radeusgd) [inline test/Visualization_Tests/src/Table_Spec.enso]: not really important as this is just tests, but our naming convention would make this
```suggestion
type Foo_Link
```

File: test/Visualization_Tests/src/Table_Spec.enso [hunk 5]
```diff
@@ -100,6 +106,11 @@ add_specs suite_builder =
             json = JS_Object.from_pairs [["json", JS_Object.from_pairs [["_display_text_", (Foo.Value 42).to_display_text],["x", 42]]]]
             vis . should_equal json.to_text
 
+        group_builder.specify "should handle datatypes with links" <|
+            vis = Visualization.prepare_visualization (FooLink.Value "test-value")
+            json = JS_Object.from_pairs [["json", JS_Object.from_pairs [["_display_text_", (FooLink.Value "test-value").to_display_text],["x", "test-value"], ["links", "[a, b, c]"]]], ["links", ["a", "b", "c"]]]
```
Comment (radeusgd) [inline test/Visualization_Tests/src/Table_Spec.enso]: I'd refactor the common subexpression.
```suggestion
            example = FooLink.Value "test-value"
            vis = Visualization.prepare_visualization example
            json = JS_Object.from_pairs [["json", JS_Object.from_pairs [["_display_text_", example.to_display_text],["x", "test-value"], ["links", "[a, b, c]"]]], ["links", ["a", "b", "c"]]]
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Connection.enso [hunk 6]
```diff
@@ -286,6 +286,11 @@ type Postgres_Connection
     save_as_data_link self destination (on_existing_file:Existing_File_Behavior = Existing_File_Behavior.Error) =
         self.data_link_setup.save_as_data_link destination on_existing_file
 
+    ## PRIVATE
+      Converts this value to a JSON serializable object.
+    to_js_object : JS_Object
+    to_js_object self =
+        JS_Object.from_pairs [["type", "SQLite_Connection"], ["links", self.connection.tables.at "Name" . to_vector]]
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Connection.enso]:
```suggestion
        JS_Object.from_pairs [["type", "Postgres_Connection"], ["links", self.connection.tables.at "Name" . to_vector]]    
```

**Review by radeusgd — APPROVED** [review]
Enso code looks good.
One important typo (connection name in Postgres) needs to be fixed.

File: app/gui2/src/components/visualizations/TableVisualization.vue (hunk 3 above)
Comment (kazcw) [inline app/gui2/src/components/visualizations/TableVisualization.vue]:
```suggestion
      return `Double click to view this ${onClickTooltipValue.value} in a separate node`
```

Radek is right, and also a typo.

File: app/gui2/src/components/visualizations/JSONVisualization/JsonObjectWidget.vue [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
 const props = defineProps<{ data: object }>()
 const emit = defineEmits<{
   createProjection: [path: (string | number)[][]]
 }>()
+const config = useVisualizationConfig()
 
 const MAX_INLINE_LENGTH = 40
+const JSON_OBJECT_TYPE = 'Standard.Base.Data.Json.JS_Object'
+
+const isClickThroughEnabled = config.nodeType === JSON_OBJECT_TYPE
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/JSONVisualization/JsonObjectWidget.vue]: This duplicates logic from JSONVisualization.

File: app/gui2/src/components/visualizations/JSONVisualization/JsonObjectWidget.vue [hunk 8]
```diff
@@ -74,4 +79,8 @@ function entryTitle(key: string) {
 .key {
   color: darkred;
 }
+.key-link {
+  color: blue;
+  text-decoration: underline;
+}
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/JSONVisualization/JsonObjectWidget.vue]: Only one class is needed for `key`; different styles can be specified for `.key` and `.viewonly .key`.

**Review by kazcw — CHANGES_REQUESTED** [review]

**Review by kazcw — APPROVED** [review]

**Pull request merged**